### ISSUE TITLE
gradient of tf.floor

### ISSUE THREAD

**Issue by ppwwyyxx** [tensorflow/tensorflow#897]
```python
import numpy as np
import tensorflow as tf
f = np.array([3.8], dtype='float32')
vif = tf.placeholder(tf.float32, shape=[1])

#out = vif + tf.cast(tf.cast(vif, tf.int32), tf.float32)
out = vif + tf.floor(vif)
grad = tf.gradients(tf.reduce_sum(out), vif)

sess = tf.Session()
sess.run(tf.initialize_all_variables())
print sess.run(grad, feed_dict={vif: f})  # print 2

import theano
import theano.tensor as T
vif = T.fvector()
out = vif + T.floor(vif)
grad = T.grad(out.sum(), vif)
func = theano.function([vif], grad)
print func(f) # print 1
```
It looks like the gradient of `tf.floor(x)` w.r.t `x` is 1. But I'm expecting it to be 0, as in theano.
Right now I could use a double casting as a work around, but why is it different?

**Comment by girving** [comment]
Yes, the gradient of floor is always zero.

**Comment by girving** [comment]
Well, almost always.

**Comment by ppwwyyxx** [comment]
Should it be as simple as changing:
```python
@ops.RegisterGradient("Floor")
def _FloorGrad(_, grad):
  return grad  # return 0
```
in `python/ops/math_grad.py`?

**Comment by girving** [comment]
Yes, changing it to return `[None]` would be the fix.

**Comment by ppwwyyxx** [comment]
Thanks. Could you explain why `None` is preferred over 0 ?
I observed that the gradient of `tf.cast(tf.cast(vif, tf.int32), tf.float32)` is also `None`, that's why I added   vif in the example code.

**Comment by girving** [comment]
`None` will be treated by the gradient code as "no connection", which is mathematically equivalent to zero but faster since it will never construct large zero matrices.  There's some related discussion here: #783.
